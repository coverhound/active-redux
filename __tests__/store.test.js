import Store from 'active-redux/store';
import createMockStore from 'fixtures/store';

describe('Store', () => {
  const mockStore = createMockStore();
  Store.bind(mockStore);

  describe('#store', () => {
    test('is the bound store', () => {
      expect(Store.store).toBe(mockStore);
    });
  });

  describe('#dispatch', () => {
    test('provides a getter for store dispatch', () => {
      expect(Store.dispatch).toBe(mockStore.dispatch);
    });
  });

  describe('#state', () => {
    test('provides a getter for store dispatch', () => {
      expect(Store.state).toBe(mockStore.getState().api.resources);
    });
  });
});
