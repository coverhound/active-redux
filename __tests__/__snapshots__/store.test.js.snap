// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store #all should return all values from a resource store 1`] = `
Array [
  Object {
    "attributes": Object {
      "body": "First!",
    },
    "id": "5",
    "links": Object {
      "self": "http://example.com/comments/5",
    },
    "relationships": Object {
      "author": Object {
        "data": Object {
          "id": "9",
          "type": "people",
        },
      },
    },
    "type": "comments",
  },
  Object {
    "attributes": Object {
      "body": "I like XML better",
    },
    "id": "12",
    "links": Object {
      "self": "http://example.com/comments/12",
    },
    "relationships": Object {
      "author": Object {
        "data": Object {
          "id": "9",
          "type": "people",
        },
      },
    },
    "type": "comments",
  },
]
`;

exports[`Store #find should find an entity in by query 1`] = `
Object {
  "attributes": Object {
    "body": "First!",
  },
  "id": "5",
  "links": Object {
    "self": "http://example.com/comments/5",
  },
  "relationships": Object {
    "author": Object {
      "data": Object {
        "id": "9",
        "type": "people",
      },
    },
  },
  "type": "comments",
}
`;

exports[`Store #find should find an entity in store by id 1`] = `undefined`;

exports[`Store #where should allow arrays as lookup value 1`] = `
Array [
  Object {
    "attributes": Object {
      "body": "First!",
    },
    "id": "5",
    "links": Object {
      "self": "http://example.com/comments/5",
    },
    "relationships": Object {
      "author": Object {
        "data": Object {
          "id": "9",
          "type": "people",
        },
      },
    },
    "type": "comments",
  },
  Object {
    "attributes": Object {
      "body": "I like XML better",
    },
    "id": "12",
    "links": Object {
      "self": "http://example.com/comments/12",
    },
    "relationships": Object {
      "author": Object {
        "data": Object {
          "id": "9",
          "type": "people",
        },
      },
    },
    "type": "comments",
  },
]
`;

exports[`Store #where should return an array of values matching a query 1`] = `
Array [
  Object {
    "attributes": Object {
      "body": "First!",
    },
    "id": "5",
    "links": Object {
      "self": "http://example.com/comments/5",
    },
    "relationships": Object {
      "author": Object {
        "data": Object {
          "id": "9",
          "type": "people",
        },
      },
    },
    "type": "comments",
  },
]
`;

exports[`Store #where should search for nested properties 1`] = `
Array [
  Object {
    "attributes": Object {
      "address": Object {
        "road": "123 Main St",
      },
      "first-name": "Dan",
      "last-name": "Gebhardt",
      "twitter": "dgeb",
    },
    "id": "9",
    "links": Object {
      "self": "http://example.com/people/9",
    },
    "relationships": Object {
      "comments": Object {
        "data": Array [
          Object {
            "id": "5",
            "type": "comments",
          },
          Object {
            "id": "12",
            "type": "comments",
          },
        ],
      },
    },
    "type": "people",
  },
]
`;
